### PR TITLE
Created a function to convert uppercase

### PR DESCRIPTION
**Pull request by abhishekraoas**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

**Review by BamaCharanChhandogi — COMMENTED** [review]
removeRmain method then add the test case

**Comment by abhishekraoas** [comment]
I updated the code according to your guidance

File: src/main/java/com/thealgorithms/strings/WordUpperCase.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.thealgorithms.strings;
+
+public class WordUpperCase {
+    public static void main(String[] args) {
+        String str = "hii my name is abhishek";
```
Comment (ylcn91) [inline src/main/java/com/thealgorithms/strings/WordUpperCase.java:5]: Rename 'str' to 'inputText' for clarity.

File: src/main/java/com/thealgorithms/strings/WordUpperCase.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package com.thealgorithms.strings;
+
+public class WordUpperCase {
```
Comment (ylcn91) [inline src/main/java/com/thealgorithms/strings/WordUpperCase.java:3]: The class name 'WordUpperCase' could be more descriptive of its purpose.
For example, 'Capitalizer' or 'StringCapitalizer'.

File: src/main/java/com/thealgorithms/strings/WordUpperCase.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    public static void main(String[] args) {
+        String str = "hii my name is abhishek";
+        System.out.println(toUpperCase(str));
+    }
+
+    public static String toUpperCase(String str) {
+        StringBuilder sb = new StringBuilder("");
```
Comment (ylcn91) [inline src/main/java/com/thealgorithms/strings/WordUpperCase.java:10]: new StringBuilder() without an argument would be cleaner.

File: src/main/java/com/thealgorithms/strings/WordUpperCase.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        if (str == null || str.isEmpty()) {
+            return str; // Handle null or empty input gracefully
+        }
+        char ch = Character.toUpperCase(str.charAt(0));
+        sb.append(ch);
```
Comment (ylcn91) [inline src/main/java/com/thealgorithms/strings/WordUpperCase.java:15]: Consider using method chaining with StringBuilder to reduce verbosity.

File: src/main/java/com/thealgorithms/strings/WordUpperCase.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        for (int i = 1; i < str.length(); i++) {
+            if (str.charAt(i) == ' ' && i < str.length() - 1) {
+                sb.append(str.charAt(i));
+                i++;
```
Comment (ylcn91) [inline src/main/java/com/thealgorithms/strings/WordUpperCase.java:19]: The logic here is a bit hard to follow due to the manual increment of 'i'.
I think we could improve readability by using a while loop or extracting this logic into a separate method.

File: src/test/java/com/thealgorithms/strings/WordUpperCaseTest.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        assertEquals(expected1, WordUpperCase.toUpperCase(input1));
+
+        String input2 = "this is a test";
+        String expected2 = "This Is A Test";
+        assertEquals(expected2, WordUpperCase.toUpperCase(input2));
+
+        String input3 = "hello";
+        String expected3 = "Hello";
+        assertEquals(expected3, WordUpperCase.toUpperCase(input3));
+    }
```
Comment (ylcn91) [inline src/test/java/com/thealgorithms/strings/WordUpperCaseTest.java:21]: Suggestions for test cases: 
- Passing an empty string.
- Passing a null value.
- A string with multiple spaces between words.
- A string with special characters and numbers.
- Case with leading and trailing spaces.
